File: test/language/subtypes/cpp98/StructureSubtypeTest.cpp
#include <cstdio>
#include <string>
#include <fstream>

#include "gtest/gtest.h"

#include "subtypes/structure_subtype/SubtypeStructure.h"
#include "subtypes/structure_subtype/Student.h"

namespace subtypes
{
namespace structure_subtype
{

TEST(StructureSubtypeTest, TestSubtype)
{
    const uint32_t identifier = 0xFFFF;
    const std::string name = "Name";
    Student student;
    student.setIdentifier(identifier);
    student.setName(name);

    SubtypeStructure subtypeStructure;
    subtypeStructure.setStudent(student);
    const Student readStudent = subtypeStructure.getStudent();

    ASSERT_EQ(student, readStudent);
}

} // namespace structure_subtype
} // namespace subtypes

File: test/language/subtypes/java/subtypes/StructureSubtypeTest.java
package subtypes;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

import subtypes.structure_subtype.SubtypeStructure;
import subtypes.structure_subtype.TestStructure;

public class StructureSubtypeTest
{
    @Test
    public void testSubtype()
    {
        final int identifier = 0xFFFF;
        final String name = "Name";
        final TestStructure testStructure = new TestStructure(identifier, name);
        final SubtypeStructure subtypeStructure = new SubtypeStructure(testStructure);
        final TestStructure readTestStructure = subtypeStructure.getStudent();
        assertEquals(testStructure, readTestStructure);
    }
}
